sorc/ncep_post.fd/MPI_FIRST.f

```diff
@@ -149,7 +157,8 @@ SUBROUTINE MPI_FIRST()
 !     counts, disps for gatherv and scatterv
 !
       do i = 0, num_procs - 1
-         call para_range(1,jm,num_procs,i,jsx,jex) 
+         call para_range2(1,jm,1,im,num_procs,i,jsx,jex,isx,iex) 
+!gwv  delete after 2D support is validated        call para_range(1,jm,num_procs,i,jsx,jex) 
```

Comment:
@GeorgeVandenberghe-NOAA Would you consider to assign i halo for ista_2l, jend_2u, ivend_2u?

Reply:
@GeorgeVandenberghe-NOAA do you want to have num_procs in i and j dimensions separately so that we will have some flexibility to control mpi tasks assigned to i and j dimension? E.g. we can have num_procs_i, num_procs_j, and num_procs=num_procs_i * num_procs_j. We have still use para_range for i dimension with num_procs_i tasks.

Reply:
> @GeorgeVandenberghe-NOAA do you want to have num_procs in i and j dimensions separately so that we will have some flexibility to control mpi tasks assigned to i and j dimension? E.g. we can have num_procs_i, num_procs_j, and num_procs=num_procs_i * num_procs_j. We have still use para_range for i dimension with num_procs_i tasks.

is this how it's done in FV3 model?

Reply:
> @GeorgeVandenberghe-NOAA do you want to have num_procs in i and j dimensions separately so that we will have some flexibility to control mpi tasks assigned to i and j dimension? E.g. we can have num_procs_i, num_procs_j, and num_procs=num_procs_i * num_procs_j. We have still use para_range for i dimension with num_procs_i tasks.

I think this is a great idea if this can be done.  It gives us flexibility to go back to decompose Y only if necessary